iroha/src/wsv.rs

```diff
@@ -118,66 +148,74 @@ impl WorldStateView {
         &self.world.trusted_peers_ids
     }
 
-    /// Returns reference for parameters
-    pub const fn parameters(&self) -> &RwLock<Vec<Parameter>> {
-        &self.world.parameters
-    }
-
     /// Get `Domain` without an ability to modify it.
+    ///
     /// # Errors
     /// Fails if there is no domain
-    pub fn domain<T>(&self, name: &str, f: impl FnOnce(&Domain) -> T) -> Result<T> {
+    pub fn domain(&self, name: &str) -> Result<DashmapRef<Name, Domain>> {
         let domain = self
             .world
             .domains
             .get(name)
             .ok_or_else(|| FindError::Domain(name.to_owned()))?;
-        Ok(f(domain.value()))
+        Ok(domain)
     }
 
-    /// Get `Account` and pass it to closure
+    /// Get `Domain` with an ability to modify it.
+    ///
+    /// # Errors
+    /// Fails if there is no domain
+    pub fn domain_mut(&self, name: &str) -> Result<DashmapRefMut<Name, Domain>> {
+        let domain = self
+            .world
+            .domains
+            .get_mut(name)
+            .ok_or_else(|| FindError::Domain(name.to_owned()))?;
+        Ok(domain)
+    }
```

Comment:
Is it okay to get mutable reference?
[Doc](https://docs.rs/dashmap/4.0.2/dashmap/struct.DashMap.html#method.get_mut) says that it may deadlock if any reference is hold

Reply:
I think they mean theoretical possibility like yes, the mutable reference with spinlocks may deadlock if we make certain conditions in our code.

Reply:
But still in any way we will take a mutable reference, it just depends how much it lives.